LoRaEngine/modules/LoRaWanNetworkSrvModule/LoraTools/LoRaMessage/LoRaPayloadJoinRequest.cs

```diff
@@ -42,22 +42,10 @@ public LoRaPayloadJoinRequest()
         {
         }
 
-        public LoRaPayloadJoinRequest(byte[] inputMessage)
-            : base(inputMessage)
-        {
-            Mhdr = new Memory<byte>(inputMessage, 0, 1);
-            // get the joinEUI field
-            AppEUI = new Memory<byte>(inputMessage, 1, 8);
-            // get the DevEUI
-            DevEUI = new Memory<byte>(inputMessage, 9, 8);
-            // get the DevNonce
-            DevNonce = DevNonce.Read(inputMessage.AsSpan(17));
-        }
-
         public LoRaPayloadJoinRequest(string appEUI, string devEUI, DevNonce devNonce)
         {
             // Mhdr is always 0 in case of a join request
-            Mhdr = new byte[1] { 0x00 };
+            MHdr = new MacHeader(0x00);
```

Comment:
I think we should use the more meaningful constructor here:

```suggestion
            MHdr = new MacHeader(MacMessageType.JoinRequest);
```